lua/pac3/core/shared/movement.lua

```diff
@@ -145,6 +151,24 @@ pac.AddHook("Move", "custom_movement", function(ply, mv)
 
 	ply:SetJumpPower(self.JumpHeight)
 
+	if SERVER then
+		if allowMass:GetInt() == 1 then
+			ply:GetPhysicsObject():SetMass(math.Clamp(self.Mass, massLowerLimit:GetFloat(), massUpperLimit:GetFloat()))
+		end
+	end
+	
+	if (movementConvar:GetInt() == 1 and massDamageScale:GetInt() == 1 or (movementConvar:GetInt() == -1 and hook.Run("PlayerNoClip", ply, true) == true)) and massDamageScale:GetInt() == 1 then
```

Comment:
(more readable form just for my sake of demonstration)
```
(
	movementConvar:GetInt() == 1
	and massDamageScale:GetInt() == 1
	
	or
	(
		movementConvar:GetInt() == -1
		and hook.Run("PlayerNoClip", ply, true) == true
	)
)
and massDamageScale:GetInt() == 1
```

You don't need to check massdamagescale twice. I would keep only the outside one because it governs everything anyway.
The expression is not wrong though.